mlflow/models/model.py

```diff
@@ -698,3 +711,103 @@ def get_model_info(model_uri: str) -> ModelInfo:
         mlflow_version=model_meta.mlflow_version,
         metadata=model_meta.metadata,
     )
+
+
+def get_model_requirements_files(resolved_uri: str) -> Dict[str, str]:
+    requirements_txt_file = None
+    conda_yaml_file = None
+    try:
+        requirements_txt_file = _download_artifact_from_uri(
+            artifact_uri=append_to_uri_path(resolved_uri, _REQUIREMENTS_FILE_NAME)
+        )
+        conda_yaml_file = _download_artifact_from_uri(
+            artifact_uri=append_to_uri_path(resolved_uri, _CONDA_ENV_FILE_NAME)
+        )
+    except Exception as ex:
+        file_name = (
+            _REQUIREMENTS_FILE_NAME if requirements_txt_file is None else _CONDA_ENV_FILE_NAME
+        )
+        raise MlflowException(
+            f'Failed to download "{file_name}" from "{resolved_uri}"',
+            RESOURCE_DOES_NOT_EXIST,
+        ) from ex
+
+    return {
+        _REQUIREMENTS_FILE_NAME: requirements_txt_file,
+        _CONDA_ENV_FILE_NAME: conda_yaml_file,
+    }
```

Comment:
I'd return a tuple of `(requirements_txt_file, conda_yaml_file)` and unpack them in the caller.

Reply:
that was my initial approach, but changed to this as i was kind of worried about having to get the order right. i'll change to return a tuple!

Reply:
/ai let use know if you have a better approach for this

Reply:
/ai That's fair. Can we use a namedtuple?